nntrainer/tensor/cache_pool.cpp

```diff
@@ -133,69 +181,111 @@ std::shared_ptr<MemoryData<float>> CachePool::getMemory(unsigned int id) {
 }
 
 void CachePool::flush() {
-  for (auto elem : actives)
-    elem->swapOut();
+  for (auto &elem : actives)
+    elem->swapOut(CacheElem::LAST_ACCESS);
+
+  for (auto &[id, elem] : elems)
+    elem->reset();
 
   actives.clear();
 }
 
 void CachePool::flushExcept(unsigned int order) {
+  auto exe_orders = getMemoryExecOrder();
+
   actives.remove_if([&, order](auto elem) -> bool {
-    auto exe_order = elem->getExeOrder();
+    auto id = elem->getId();
+    auto exe_order = exe_orders.at(id - 1);
     auto found = std::find(exe_order.begin(), exe_order.end(), order);
     if (found == exe_order.end()) {
-      elem->swapOut();
+      /**
+       * We assumes that flushExcept will be called in front of each execution
+       * order, and the order is incremental. So, we can conclude that, if the
+       * order passes by the max order of the cache element, it was LAST access
+       * of the element.
```

Comment:
For now, it is ok. However, in extreme optimization cases, we know and can save the list that needs to be flushed after the first iteration so that we can memorize them and use them until the training end without finding them again.

Reply:
Thanks. I'll consider it in later.